### ISSUE TITLE
Cannot get privileged mode working properly since v4.20.0

### ISSUE THREAD

**Issue by lekseven** [docker/for-mac#7062]
### Description

I have a big complex `entrypoint.sh` file which utilizes `chroot` to get access to the host docker instance. It looks like this:

```shell
#!/bin/sh

set -e

# some code ...

chroot /host_root docker <any command>

# some code ...
```

where `/host_root` mirrors into the host root `/` via `docker-compose.yml`:

```yml
version: "3.8"

services:
    mounter:
        build:
            context: ./
            dockerfile: dnsmasq.Dockerfile
            network: host
        privileged: true
        restart: unless-stopped
        labels:
            kind: system
        ports:
            - '127.0.0.1:553:53/udp'
        volumes:
            - /:/host_root

```

`dnsmasq.Dockerfile`
```Dockerfile
FROM alpine

RUN apk add --no-cache \
    dnsmasq iptables

COPY entrypoint.sh /entrypoint.sh

ENTRYPOINT ["/entrypoint.sh"]
```

It used to work perfectly for several years, but since Docker Desktop v4.20.0 (and newer) the container started crushing with the following error:

`chroot: can't execute 'docker': No such file or directory`

I tried to change different settings in Docker Desktop (e.g. sharing implementation, virtualization, rosetta, etc.) but with no effect. I would appreciate any ideas on how to solve this problem because I haven't been able to update Docker for Mac to a newer version for a year. I'm stuck on version 4.19.0 due to this issue.

### Reproduce

1. Configure Docker Desktop for Mac on an Apple Silicon host.
2. Create minimal files from the description above.
3. Run `docker compose up`

### Expected behavior

No error.

### docker version

```bash
Client:
 Cloud integration: v1.0.35+desktop.5
 Version:           24.0.6
 API version:       1.43
 Go version:        go1.20.7
 Git commit:        ed223bc
 Built:             Mon Sep  4 12:28:49 2023
 OS/Arch:           darwin/arm64
 Context:           desktop-linux

Server: Docker Desktop 4.25.0 (126437)
 Engine:
  Version:          24.0.6
  API version:      1.43 (minimum version 1.12)
  Go version:       go1.20.7
  Git commit:       1a79695
  Built:            Mon Sep  4 12:31:36 2023
  OS/Arch:          linux/arm64
  Experimental:     false
 containerd:
  Version:          1.6.22
  GitCommit:        8165feabfdfe38c65b599c4993d227328c231fca
 runc:
  Version:          1.1.8
  GitCommit:        v1.1.8-0-g82f18fe
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

### docker info

```bash
Client:
 Version:    24.0.6
 Context:    desktop-linux
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.11.2-desktop.5
    Path:     /Users/lekseven/.docker/cli-plugins/docker-buildx
  compose: Docker Compose (Docker Inc.)
    Version:  v2.23.0-desktop.1
    Path:     /Users/lekseven/.docker/cli-plugins/docker-compose
  dev: Docker Dev Environments (Docker Inc.)
    Version:  v0.1.0
    Path:     /Users/lekseven/.docker/cli-plugins/docker-dev
  extension: Manages Docker extensions (Docker Inc.)
    Version:  v0.2.20
    Path:     /Users/lekseven/.docker/cli-plugins/docker-extension
  init: Creates Docker-related starter files for your project (Docker Inc.)
    Version:  v0.1.0-beta.9
    Path:     /Users/lekseven/.docker/cli-plugins/docker-init
  sbom: View the packaged-based Software Bill Of Materials (SBOM) for an image (Anchore Inc.)
    Version:  0.6.0
    Path:     /Users/lekseven/.docker/cli-plugins/docker-sbom
  scan: Docker Scan (Docker Inc.)
    Version:  v0.26.0
    Path:     /Users/lekseven/.docker/cli-plugins/docker-scan
  scout: Docker Scout (Docker Inc.)
    Version:  v1.0.9
    Path:     /Users/lekseven/.docker/cli-plugins/docker-scout

Server:
 Containers: 26
  Running: 13
  Paused: 0
  Stopped: 13
 Images: 180
 Server Version: 24.0.6
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 8165feabfdfe38c65b599c4993d227328c231fca
 runc version: v1.1.8-0-g82f18fe
 init version: de40ad0
 Security Options:
  seccomp
   Profile: unconfined
  cgroupns
 Kernel Version: 6.4.16-linuxkit
 Operating System: Docker Desktop
 OSType: linux
 Architecture: aarch64
 CPUs: 8
 Total Memory: 8.729GiB
 Name: linuxkit-62d62dfa3ab8
 ID: 251bbafb-8b58-432b-ac90-cacbdba31295
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 HTTP Proxy: http.docker.internal:3128
 HTTPS Proxy: http.docker.internal:3128
 No Proxy: hubproxy.docker.internal
 Experimental: false
 Insecure Registries:
  hubproxy.docker.internal:5555
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: daemon is not using the default seccomp profile
```

### Diagnostics ID

0552F070-E49A-414B-811C-FE8B54FCE26F/20231107171914

**Comment by lekseven** [comment]
Turned out the host root directory is not fully mounted to the container (permission denied from inside the container) so that this part stopped working after the update:

```yml
volumes:
    - /:/host_root
```

**Comment by lekseven** [comment]
I had to to re-write the entrypoint script without using chroot to get it work with Docker > 4.19.

**Comment by dgageot** [comment]
Hi @lekseven, the fix should ship in Docker Desktop 4.26.0. In the mean time, disabling Rosetta should fix it. But it'll be slower.

Do you have a minimal repro procedure that I can follow to ensure that it's fixed on our side?

**Comment by dgageot** [comment]
I'mn going to close this issue since Docker Desktop 4.26.0 should fix it. Feel free to reopen if you still have the issue.
